src/FormattedMessage.js

```diff
@@ -0,0 +1,39 @@
+const format = (ctx, values) => ctx.parent.$formatMessage(ctx.props, values)
+const placeholder = (name) => `@\0@\0${name}\0@\0@`
+const placeholderRegex = /@\0@\0(.*?)\0@\0@/g
+
+export default {
+    functional: true,
+
+    props: {
+        id: {type: String, required: true},
+        defaultMessage: String,
+        values: Object,
```

Comment:
It's not about scopes. I'm actually passing the object. In the demo example, it's an inline object, but often it will be something from data or generated from other sources. There is no point in using any form of bindings to do that. I'd say that slots are a stretch already, but there is no real way around that.